scripts.js

```diff
@@ -79,6 +93,11 @@ async function fetchDataDocuments() {
 
         options.year.prepend(option);
     }
+
+    for (let statsObj of tabletop.sheets('stats').all()) {
+        stats.set(statsObj['Year'], statsObj);
+        delete statsObj['Year'];
```

Comment:
Why is this necessary?

Reply:
Normally JavaScript should handle this kind of operation automatically, is there a specific memory concern here?

Reply:
Totally reworking this actually. The whole for loop is quite unnecessary.